client/web/src/user/settings/auth/UserSettingsSecurityPage.tsx

```diff
@@ -48,319 +48,257 @@ interface Props {
-                            </Form>
-                        </Container>
-                    </>
-                )}
-            </>
-        )
-    }
+        setAccounts({ fetched: data?.user?.externalAccounts.nodes, lastRemoved: '' })
+    }, [data])
 
-    private onAccountRemoval = (removeId: string, name: string): void => {
+    const onAccountRemoval = (removeId: string, name: string): void => {
         // keep every account that doesn't match removeId
-        this.setState(previousState => ({
-            accounts: {
-                fetched: previousState.accounts.fetched?.filter(({ id }) => id !== removeId),
-                lastRemoved: name,
-            },
-        }))
+        setAccounts({ fetched: accounts.fetched?.filter(({ id }) => id !== removeId), lastRemoved: name })
     }
 
-    private onOldPasswordFieldChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
-        this.setState({ oldPassword: event.target.value })
+    const onOldPasswordFieldChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        setOldPassword(event.target.value)
     }
 
-    private onNewPasswordFieldChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
-        this.setState({ newPassword: event.target.value }, () => this.validateForm())
+    const onNewPasswordFieldChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        setNewPassword(event.target.value)
+        validateForm()
     }
 
-    private onNewPasswordConfirmationFieldChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
-        this.setState({ newPasswordConfirmation: event.target.value }, () => this.validateForm())
+    const onNewPasswordConfirmationFieldChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        setNewPasswordConfirmation(event.target.value)
+        validateForm()
     }
 
-    private validateForm(): void {
-        if (this.newPasswordConfirmationField) {
-            if (this.state.newPassword === this.state.newPasswordConfirmation) {
-                this.newPasswordConfirmationField.setCustomValidity('') // valid
+    function validateForm(): void {
+        if (newPasswordConfirmationField) {
+            if (newPassword === newPasswordConfirmation) {
+                newPasswordConfirmationField.setCustomValidity('') // valid
             } else {
-                this.newPasswordConfirmationField.setCustomValidity("New passwords don't match.")
+                newPasswordConfirmationField.setCustomValidity("New passwords don't match.")
             }
         }
     }
 
-    private handleSubmit = (event: React.FormEvent<HTMLFormElement>): void => {
-        this.submits.next(event)
+    const [updatePassword] = useMutation<UpdatePasswordResult, UpdatePasswordVariables>(UPDATE_PASSWORD, {
+        variables: {
+            oldPassword,
+            newPassword,
+        },
+        onError: error => {
+            handleError(error)
+        },
```

Comment:
First-class functions, baby

```suggestion
        onError: handleError,
```